site/client/blocks/ProminentLink/ProminentLink.tsx

```diff
@@ -0,0 +1,112 @@
+import { computed } from "mobx"
+import {
+    strToQueryParams,
+    queryParamsToStr,
+    splitURLintoPathAndQueryString,
+    QueryParams
+} from "utils/client/url"
+import { EntityUrlBuilder } from "charts/ChartUrl"
+import { union, isEmpty } from "charts/Util"
+
+const PROMINENT_LINK_CLASSNAME = "wp-block-owid-prominent-link"
+
+@observer
+class ProminentLink extends React.Component<{
+    originalURL: string
+    innerHTML: string | null
+}> {
+    @computed get originalURLPath() {
+        return splitURLintoPathAndQueryString(this.props.originalURL).path
+    }
+
+    @computed get originalURLQueryString(): string | undefined {
+        return splitURLintoPathAndQueryString(this.props.originalURL)
+            .queryString
+    }
+
+    @computed get originalURLQueryParams(): QueryParams | undefined {
+        const { originalURLQueryString } = this
+
+        return originalURLQueryString
+            ? strToQueryParams(originalURLQueryString)
+            : undefined
+    }
+
+    @computed get originalURLEntityCodes(): string[] {
+        const originalEntityQueryParam = this.originalURLQueryParams?.[
+            "country"
+        ]
+
+        const entityQueryParamExists =
+            originalEntityQueryParam != undefined &&
+            !isEmpty(originalEntityQueryParam)
+
+        return entityQueryParamExists
+            ? EntityUrlBuilder.queryParamToEntities(originalEntityQueryParam!)
+            : []
+    }
+
+    @computed get entitiesInGlobalEntitySelection(): string[] {
+        return Grapher.globalEntitySelection.selectedEntities.map(
+            entity => entity.code
+        )
+    }
+
+    @computed get updatedEntityQueryParam(): string {
+        const newEntityList = union(
+            this.originalURLEntityCodes,
+            this.entitiesInGlobalEntitySelection
+        )
+
+        return EntityUrlBuilder.entitiesToQueryParam(newEntityList)
+    }
+
+    @computed get updatedURLParams(): QueryParams {
+        const { originalURLQueryParams, updatedEntityQueryParam } = this
+
+        return {
+            ...originalURLQueryParams,
+            ...(!isEmpty(updatedEntityQueryParam) && {
+                country: updatedEntityQueryParam
+            })
+        }
+    }
+
+    @computed get updatedURL() {
+        return this.originalURLPath + queryParamsToStr(this.updatedURLParams)
+    }
+
+    render() {
+        return (
+            <a
+                dangerouslySetInnerHTML={{ __html: this.props.innerHTML ?? "" }}
+                href={this.updatedURL}
+            />
+        )
+    }
+}
+
+export const render = () => {
+    document
+        .querySelectorAll<HTMLElement>(`.${PROMINENT_LINK_CLASSNAME}`)
+        .forEach(el => {
+            const anchorTag = el.querySelector("a")
+            if (!anchorTag) return
+
+            const originalUrl = anchorTag.href
+            const innerHTML = anchorTag.innerHTML
+
+            const rendered = (
+                <ProminentLink
```

Comment:
Should this be a React component, or would it suffice to have a find/replace of the `href` attribute with entities inserted into query string? If we replace the full `a` tag, what happens if authors had specified other attributes on the a tag?

Reply:
It's a React component so it can react dynamically to changes in the floating country selector.

Ah, good point! I can bring over all attributes on the anchor tag, plus the original div.

Reply:
Ah gotcha, forgot for a second that this was dynamic!

Reply:
Yes preserving existing attributes will probably be important.